tests/analyze_results.py

```diff
@@ -0,0 +1,129 @@
+import json
+import collections
+import functools
+import re
+import argparse
+
+
+@functools.total_ordering
+class RuleResult(object):
```

Comment:
if we are using this for loading the json I'd suggest to also use it for creating it. Otherwise it will be difficult to keep both in sync.